Create a CODE_OF_CONDUCT.md using GitHub's default template (unmodified) (#9805)
endless-sky/endless-sky: .github/CODE_OF_CONDUCT.md

interaction with the people involved, including unsolicited interaction with
those enforcing the Code of Conduct, for a specified period of time. This
includes avoiding interactions in community spaces as well as external channels
like social media (inc. Discord). Violating these terms may lead to a temporary or
permanent ban.

### 3. Temporary Ban

**Community Impact**: A serious violation of community standards, including
sustained inappropriate behavior.

**Consequence**: A temporary ban from any sort of interaction or public
communication with the community for a specified period of time. No public or
private interaction with the people involved, including unsolicited interaction
with those enforcing the Code of Conduct, is allowed during this period.
Violating these terms may lead to a permanent ban.

### 4. Permanent Ban

**Community Impact**: Demonstrating a pattern of violation of community
standards, including sustained inappropriate behavior,  harassment of an
individual, or aggression toward or disparagement of classes of individuals.

**Consequence**: A permanent ban from any sort of public interaction within
the community.

## Attribution

> This Code of Conduct is adapted from the [Contributor Covenant][homepage],

Reviewer: What do these links do? Oughtn't they be removed if they don't do anything?
```suggestion
This Code of Conduct is adapted from the Contributor Covenant homepage,
```

version 2.0, available at
https://www.contributor-covenant.org/version/2/0/code_of_conduct.html.

Community Impact Guidelines were inspired by [Mozilla's code of conduct
enforcement ladder](https://github.com/mozilla/diversity).

> [homepage]: https://www.contributor-covenant.org

Reviewer: ```suggestion
homepage: https://www.contributor-covenant.org
```

  address, without their explicit permission
* Other conduct which could reasonably be considered inappropriate in a
  professional setting

## Enforcement Responsibilities

Community leaders are responsible for clarifying and enforcing our standards of
acceptable behavior and will take appropriate and fair corrective action in
response to any behavior that they deem inappropriate, threatening, offensive,
or harmful.

Community leaders have the right and responsibility to remove, edit, or reject
comments, commits, code, wiki edits, issues, and other contributions that are
not aligned to this Code of Conduct, and will communicate reasons for moderation
decisions when appropriate.

## Scope

This Code of Conduct applies within all community spaces, and also applies when
an individual is officially representing the community in public spaces.
Examples of representing our community include using an official e-mail address,
posting via an official social media account, or acting as an appointed
representative at an online or offline event.

## Enforcement

Instances of abusive, harassing, or otherwise unacceptable behavior may be

> reported to the community leaders responsible for enforcement at
> https://discord.gg/ZeuASSx.

Reviewer 2: ```suggestion
reported to the community leaders responsible for enforcement via
their emails as accessed via their github profile or as listed in the
Endless Sky repository, such as the copyright and commit logs.
```

This is a GitHub community code of conduct. It should not require people to join another 3rd party platform just to file a report. Specific emails could be listed here, or directions to specifically where to find them. Either would be considerably more useful than linking to a discord server.

Reviewer 3: > My immediate thought is that some sort of Google form would be a good option. As far as I know, you don't even need a Google account to submit a response to a Google form (unless you wish to include uploaded files).

This is correct; yes, assuming you don't need to upload files, a Google form can be submitted anonymously. 

Another advantage to a Google form is that the form submissions can be shared among a group of people, so anyone whose been added to the form can see responses to it, whereas with email or Discord (if they choose to DM), whichever individual receives the email/DM could ignore it if they don't feel it's worth acting upon. In contrast, with a Google form, at the very least, a team of people could be made aware of it, making things much less likely to get ignored.

Reviewer 2: I rank email considerably above Discord because Discord is a singular entity (as is google); whereas there are millions of email providers (or even self-hosted) to suit every whim. 

A google form would definitely be much better than the discord link. That being said, just straight up listing everyone who can take appropriate action here with their particular preferred method of contact would be reasonable, too.

The downside with the google form is "what if the person is reporting a problem with someone who has access to the place where that google form goes?"

Reviewer 4: > I rank email considerably above Discord because Discord is a singular entity (as is google); whereas there are millions of email providers (or even self-hosted) to suit every whim.

I'm not sure why this makes Dsicord or Google forms an undesirable means of communication here. Would you elaborate?

Reviewer 5: Email is not a third party service but an internet protocol, this is an important difference, at least to people like me who value privacy and maybe don't want to use services provided by big companies (and only these companies) such as google or discord.

This is not to say those should not be listed, but in no way should those be the only ones listed.

Reviewer 2: RisingLeaf nailed it. Discord has a monopoly on itself. If someone objects to something in the ToS, then they are barred from using it at all. Blocked out entirely. If someone doesn't like google's ToS, they can go use some other provider's email service. Zero bar to entry. Take facebook, as an example. Lots of groups on there, but one has to register with a real name, and often provide a driver's license or phone number connected to name in order to register. If someone objects to that, then they're locked out. There's no recourse or other option.

Circling back to the topic at hand, this is a GitHub code of Conduct, primarily applicable to actions and communication here. GitHub already requires an email, so that is likely something that everyone active here already has and is familiar with. It's known technology for people active here. If they want to create a new email for communication here's purposes, they are capable of doing so, or not, as they prefer.

As such, email has the lowest bar to entry for the situation at hand, and has the least problems in terms of privacy and 3rd party restrictions.

Reviewer 6: Naev gives the option of reporting violations via Discord and email.

I don't see why we can't have both an email and a google form. That seems like those two combined would cover most people. There should be a warning if using a spam filter that email addresses from less common providers may be sent to spam by the server, because that does happen, but other than that I don't see the issue of giving multiple options.

Author: Setting up a collaborative inbox with Google Groups isn't too hard, though there's a bit of structural overhead.
If people find that desirable I can set one up.

Reviewer 3: > Setting up a collaborative inbox with Google Groups isn't too hard, though there's a bit of structural overhead. If people find that desirable I can set one up.

A shared email inbox, alongside a Google form shared with the same group of individuals, would be the ideal solution, assuming that the shared inbox works how I envision it would.
That solution would allow an individual to use email to avoid using a third-party service or submit an anonymous form response if they aren't comfortable sharing their email address.
It also avoids the drawbacks of relying on a single point of contact as the only person responsible for receiving, communicating, and acting upon complaints. 

Do you happen to know if a shared inbox would allow individuals with access to that shared inbox to modify/delete incoming complaints in such a way that it wouldn't be traceable by the other individuals with access to that inbox?

Reviewer 6: Depends on the shared inbox I think. For [Microsoft Outlook](https://support.microsoft.com/en-us/office/open-and-use-a-shared-mailbox-in-outlook-d94a8e9e-21f1-4240-808b-de9c9c088afd), it seems like only those who could delete incoming emails would be admins of Microsoft's server. However, for [Gmail](https://support.google.com/googlecloud/answer/10636599?hl=en), "delegates" would be able to delete incoming emails, so mods would have to have something like POP3 set up to download emails as they arrive, so that they would have a local copy in case that happens.

> All complaints will be reviewed and investigated promptly and fairly.

Reviewer 7: What guarantee is there that the investigation will be fair? Does someone being investigated have the right to express a lack of confidence in the moderation team? How will we go about preventing that lack of confidence in the first place?

Author: Refer to the multiple contact options conversation.

Reviewer 7: I think the contact options conversation has sort of touched on it, but addressing lack of confidence in one specific moderator. While that is certainly an issue, I think we also have to have some sort of way to assuage a lack of confidence in the entire team. It is pretty much impossible that we will ever have an entire mod team that is biased and corrupt, but that won't stop people who have been accused or are being investigated from making those allegations or just having a bad feeling toward all our mods.

All community leaders are obligated to respect the privacy and security of the

> reporter of any incident.

Reviewer 7: What about the person being investigated? Obviously, our mods won't go around leaking the private information of anyone they suspect of being uncivil, but under the code of conduct as it currently stands that would be okay.

Author: ```suggestion
reporter and reportee of any incident.
```

Reviewer 7: Maybe "All community leaders are obligated to respect the privacy and security of anyone involved in an incident, including the reporter" instead, since reportee is a slightly confusing niche word that both points to one specific person (failing to cover all possible circumstances) and might be misunderstood by someone reading the code of conduct.

## Enforcement Guidelines

Community leaders will follow these Community Impact Guidelines in determining
the consequences for any action they deem in violation of this Code of Conduct:

### 1. Correction

**Community Impact**: Use of inappropriate language or other behavior deemed
unprofessional or unwelcome in the community.

**Consequence**: A private, written warning from community leaders, providing
clarity around the nature of the violation and an explanation of why the
behavior was inappropriate. A public apology may be requested.

### 2. Warning

**Community Impact**: A violation through a single incident or series
of actions.

**Consequence**: A warning with consequences for continued behavior. No
interaction with the people involved, including unsolicited interaction with
those enforcing the Code of Conduct, for a specified period of time. This
includes avoiding interactions in community spaces as well as external channels

> like social media. Violating these terms may lead to a temporary or

Reviewer 7: Is this a case of ES-related social media, or if I'm harassing someone outside of ES and it spills over into the ES community does it become a problem to be resolved by the ES mod team?

Author: In this case it's a matter of "being awful on the discord justifies cutting you out of the Github"

Author: ```suggestion
like social media (inc. Discord). Violating these terms may lead to a temporary or
```

permanent ban.

### 3. Temporary Ban

**Community Impact**: A serious violation of community standards, including
sustained inappropriate behavior.

**Consequence**: A temporary ban from any sort of interaction or public
communication with the community for a specified period of time. No public or
private interaction with the people involved, including unsolicited interaction
with those enforcing the Code of Conduct, is allowed during this period.
Violating these terms may lead to a permanent ban.

### 4. Permanent Ban

**Community Impact**: Demonstrating a pattern of violation of community
standards, including sustained inappropriate behavior,  harassment of an
individual, or aggression toward or disparagement of classes of individuals.

**Consequence**: A permanent ban from any sort of public interaction within

> the community.

Reviewer 7: This doesn't really feel clear to me.

Temporary ban says:
- if you ignore the ban and try to interact with members of the community, especially in a negative way, you might receive a permanent ban.

If they've managed to break free from their temporary ban, though, how are you going to enforce a permanent ban? Obviously, this probably involves the offender getting banned from the Discord, blocked from interacting with the GitHub page or the dev team, etc. But if they've already circumvented the temporary ban (which appears to be exactly the same as the permanent ban, just temporary), what's going to stop them from doing the same here? Is there going to be some action taken with outside support from GitHub or Discord?

I've got even more questions, actually:
- What even constitutes "public interaction?"
- Who among the moderators has the authority to turn a temporary ban into a permanent one? If moderator X is being "harassed" by temp-banned user Y when really Y just has poor communication skills/didn't properly read the code of conduct and is trying to get their ban appealed, should moderator X be able to execute this somewhat drastic step? Is a team of two or three moderators needed?
- The criteria for a permanent ban seem to be broader and no more serious than those for a temporary ban. The only difference is the "serious violation" vs "pattern of violation." What makes the permanent ban more severe beyond it being applied more sparingly and at the moderators' discretion?

Author: Public interaction would in this case mean posting or commenting on things in the repo.

I would say that whichever moderator applies the temporary ban is not allowed to be the same moderator who applies the permanent on. That's probably as much insurance as we can reasonably build into that though.

As for what makes it more severe?
It really just comes down to whether a person is actually capable of taking on board feedback and trying to do something about their behaviour. The temporary ban serves as a warning, the permanent ban is the decision for when the warning has no effect.

Reviewer 6: Those explanations should probably be added to the code of conduct in some way.
